### PR TITLE
API diff between .NET 8 Preview 5 and .NET 8 Preview 6

### PR DESCRIPTION
**Pull request by carlossanlop**
**Known issues**:
- The AsmDiff tool sometimes skips adding `{` or `{}`. Please add a GitHub suggestion to correct it. [arcade#10981](https://github.com/dotnet/arcade/issues/10981)
- The AsmDiff tool shows APIs with decorating attribute changes as removed and re-added, instead of showing only the API. Please ignore this, it might be confusing but technically it's not incorrect. I have an issue to track this: [arcade#13814](https://github.com/dotnet/arcade/issues/13814) 
  
Repos area owners:

- ASP.NET - @dotnet/aspnet-api-review 
- WinForms - @merriemcgaw @dreddy-work @JeremyKuhne
- WPF - @singhashish-wpf and @pchaurasia14
- Libraries - @jeffhandley @karelz @ericstj @stephentoub

Libraries area owners:
 
- System @dotnet/area-system-runtime 
- System.Diagnostics.CodeAnalysis no area label, tagging most recent modifiers: @RussKie @stephentoub 
- System.Diagnostics.Metrics @tommcdon @noahfalk 
- System.Diagnostics.Tracing @tommcdon @noahfalk @tarekgh 
- System.Formats.Asn1 @dotnet/area-system-formats-asn1 
- System.IO.Compression @dotnet/area-system-io-compression 
- System.IO.Pipes No area label, tagging most recent modifiers: @MihaZupan 
- System.Net @dotnet/ncl 
- System.Numerics @dotnet/area-system-numerics 
- System.Runtime.CompilerServices: @dotnet/area-system-runtime-compilerservices 
- System.Runtime.InteropServices @dotnet/interop-contrib 
- System.Runtime.Intrinsics @dotnet/area-system-runtime-intrinsics 
- System.Security @dotnet/area-system-security 
- System.Text.Json @dotnet/area-system-text-json 
- System.Threading.Tasks @dotnet/area-system-threading-tasks 

- Microsoft.Extensions.Caching @dotnet/area-extensions-caching 
- Microsoft.Extensions.DependencyInjection @dotnet/area-extensions-dependencyinjection 
- Microsoft.Extensions.Diagnostics.Metrics @tommcdon @noahfalk 
- Microsoft.Extensions.Hosting @dotnet/area-extensions-hosting 
- Microsoft.Extensions.Logging @dotnet/area-extensions-logging 
- Microsoft.Extensions.Options @dotnet/area-extensions-options 
- Microsoft.Extensions.Primitives @dotnet/area-extensions-primitives

### CODE REVIEW

File: release-notes/8.0/preview/api-diff/preview6/Microsoft.NETCore.App/8.0-preview6_System.Numerics.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
+# System.Numerics
```
Comment (tannergooding) [inline release-notes/8.0/preview/api-diff/preview6/Microsoft.NETCore.App/8.0-preview6_System.Numerics.md:1]: `System.Numerics` LGTM, we removed the `where T : struct` constraint

File: release-notes/8.0/preview/api-diff/preview6/Microsoft.NETCore.App/8.0-preview6_System.Runtime.Intrinsics.Wasm.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
+# System.Runtime.Intrinsics.Wasm
```
Comment (tannergooding) [inline release-notes/8.0/preview/api-diff/preview6/Microsoft.NETCore.App/8.0-preview6_System.Runtime.Intrinsics.Wasm.md:1]: `System.Runtime.Intrinsics.Wasm` LGTM

CC. @radekdoulik

File: release-notes/8.0/preview/api-diff/preview6/Microsoft.NETCore.App/8.0-preview6_System.Runtime.Intrinsics.X86.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
+# System.Runtime.Intrinsics.X86
```
Comment (tannergooding) [inline release-notes/8.0/preview/api-diff/preview6/Microsoft.NETCore.App/8.0-preview6_System.Runtime.Intrinsics.X86.md:1]: `System.Runtime.Intrinsics.X86` LGTM

CC. @dotnet/avx512-contrib

File: release-notes/8.0/preview/api-diff/preview6/Microsoft.NETCore.App/8.0-preview6_System.Runtime.Intrinsics.md [hunk 4]
```diff
@@ -0,0 +1,765 @@
+# System.Runtime.Intrinsics
```
Comment (tannergooding) [inline release-notes/8.0/preview/api-diff/preview6/Microsoft.NETCore.App/8.0-preview6_System.Runtime.Intrinsics.md:1]: `System.Runtime.Intrinsics` LGTM, we removed the `where T : struct` constraint

File: release-notes/8.0/preview/api-diff/preview6/Microsoft.NETCore.App/8.0-preview6_System.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
+# System
```
Comment (tannergooding) [inline release-notes/8.0/preview/api-diff/preview6/Microsoft.NETCore.App/8.0-preview6_System.md:1]: `System` LGTM

**Review by tannergooding — APPROVED** [review]
All the area's I'm responsible for LGTM

File: release-notes/8.0/preview/api-diff/preview6/Microsoft.WindowsDesktop.App/8.0-preview6_System.ComponentModel.Design.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
++        protected internal virtual object? CreateInstance(Type type);
+-        public INestedContainer CreateNestedContainer(IComponent owningComponent, string containerName);
++        public INestedContainer CreateNestedContainer(IComponent owningComponent, string? containerName);
+-        public object GetService(Type serviceType);
++        public object? GetService(Type serviceType);
+     }
+     public sealed class DesignSurfaceCollection : ICollection, IEnumerable {
+-        int System.Collections.ICollection.Count { get; }
++        int ICollection.Count { get; }
+-        bool System.Collections.ICollection.IsSynchronized { get; }
```
Comment (dreddy-work) [inline release-notes/8.0/preview/api-diff/preview6/Microsoft.WindowsDesktop.App/8.0-preview6_System.ComponentModel.Design.md:40]: @carlossanlop, i still see this namespace optimizations as change in API.

**Review by dreddy-work — COMMENTED** [review]
WinForms is good.

File: release-notes/8.0/preview/api-diff/preview6/Microsoft.NETCore.App/8.0-preview6_System.Runtime.InteropServices.Marshalling.md
Comment (jkoritzinsky) [inline release-notes/8.0/preview/api-diff/preview6/Microsoft.NETCore.App/8.0-preview6_System.Runtime.InteropServices.Marshalling.md:1]: System.Runtime.InteropServices.Marshalling LGTM

**Review by jkoritzinsky — APPROVED** [review]

**Comment by tarekgh** [comment]
`System.Diagnostics.Metrics` LGTM. 

@davmason can you please have a look at `System.Diagnostics.Tracing`?

**Comment by davmason** [comment]
System.Diagnostics.Tracing looks good

**Review by eiriktsarpalis — APPROVED** [review]
Json changes LGTM.

**Pull request merged**